Projects/hGraphQL/Sources/Models/ContractModels.swift

```diff
@@ -80,7 +79,93 @@ public struct Contract: Codable, Hashable, Equatable {
     public let typeOfContract: TypeOfContract
     public let upcomingAgreementsTable: DetailAgreementsTable
     public let currentAgreementsTable: DetailAgreementsTable?
-    public let gradientOption: GradientOption?
+    public var gradientOption: GradientOption? {
+        if self.currentAgreement?.status == .terminated {
+            return nil
+        }
+
+        switch self.typeOfContract {
```

Comment:
you can extract this on enum level like
var gradientOption: GradientOption = {
swithc self {

}
}